### ISSUE TITLE
All devices keep ringing except on the device that answered the call, never stop ringing until manually told to

### ISSUE THREAD

**Issue by BloodyIron** [vector-im/element-android#8178]
### Steps to reproduce

1. I'm logged into multiple devices, android, desktop, whatever
2. Person calls me, and I answer on one of them
3. All other devices (generally most noticed on Android devices) keep on ringing until the end of the universe, or the battery dies, whichever comes first

#### What did you expect?
All other devices to stop ringing and know that the call has been answered

#### What happened instead?
Every device keeps ringing

### Your phone model

Any device

### Operating system version

Multiple Android versions, Linux desktop OS

### Application version and app store

Has been going on for years, so current, and many versions back

### Homeserver

Public matrix.org

### Will you send logs?

No

### Are you willing to provide a PR?

No

**Comment by MadLittleMods** [comment]
👍 Also experienced this by starting a call from a test user on Element Web, answering on another Element Web session, but the Element Android app keeps with incoming call ringing forever. Even after the call ends, I have to manually decline on Element Android to make it stop.

Here are rageshake logs if it even matters: https://github.com/matrix-org/element-android-rageshakes/issues/50619

<img src="https://user-images.githubusercontent.com/558581/225194197-ad37b202-cd60-4309-aca9-e1647f019729.png" width="150">

**Comment by systemoperator** [comment]
I have the same problem. It's extremely annoying. If e.g. one of the devices is in another room, this rings/vibrates forever until the battery is empty. There is no workaround. :disappointed:

**Comment by waclaw66** [comment]
I experience this very often, same symptoms as above. Nobody can fix it?

**Comment by BloodyIron** [comment]
Yeah this is tagged as frequent and major, but have seen nothing done on this matter since Feb... what's going on here?

**Comment by BloodyIron** [comment]
I just came home to find that my tablet was still ringing... **FOUR HOURS** after I had already answered and completed the Element VoIP call on another device away from my home. This burned about 40% of my tablet's device right before a weekend vacation travel where I really do need the battery.

This is tagged as Frequent and Major... yet I don't see any actual action on this issue. What's going on here??? >:|

**Comment by YamatoRyou** [comment]
> I just came home to find that my tablet was still ringing... **FOUR HOURS** after I had already answered and completed the Element VoIP call on another device away from my home. This burned about 40% of my tablet's device right before a weekend vacation travel where I really do need the battery.
> 
> This is tagged as Frequent and Major... yet I don't see any actual action on this issue. What's going on here??? >:|

This issue has been around for a long time but has not been solved by the development team or contributors. I tried to solve it through some indirect methods. If you are interested please see: https://github.com/YamatoRyou/element-android-incoming-call-fix- for-macrodroid#english

**Comment by bmarty** [comment]
There are multiple root causes:
- technical cause (spec): a Push is sent for `m.call.invite`, so the device can start ringing. But there is no push for `m.call.answer` or `m.call.reject`, so the device cannot know if the call has been answered elsewhere. If another event is triggering a push, the device sync and the device stops ringing. Anyway, even if pushes were receive for such events, it would not be enough. The connectivity can be lost, and in this case, the device will never stop ringing.
- implementation: the call should just not ring forever. An internal timeout (one minute) have to be implemented.

So the solution is to limit by design the duration of the ringing call. the call can stop ringing earlier, if the `m.call.answer/reject` is detected, like today.

**Comment by bmarty** [comment]
Actually for point 1, there is [this code](https://github.com/vector-im/element-android/blob/develop/vector/src/main/java/im/vector/app/features/call/webrtc/WebRtcCallManager.kt#L378) which mitigate the lack of Push. The network can still be lost.

**Comment by YamatoRyou** [comment]
> There are multiple root causes:
> 
> * technical cause (spec): a Push is sent for `m.call.invite`, so the device can start ringing. But there is no push for `m.call.answer` or `m.call.reject`, so the device cannot know if the call has been answered elsewhere. If another event is triggering a push, the device sync and the device stops ringing. Anyway, even if pushes were receive for such events, it would not be enough. The connectivity can be lost, and in this case, the device will never stop ringing.
> * implementation: the call should just not ring forever. An internal timeout (one minute) have to be implemented.
> 
> So the solution is to limit by design the duration of the ringing call. the call can stop ringing earlier, if the `m.call.answer/reject` is detected, like today.

I have been troubled by this problem for a long time. Based on my long-term observations (user perspective), I have drawn some conclusions:
## Occasion 1:
1. Make the Element appear on the screen;
2. Press the Home button (optional);
3. Press the power button to turn off and lock the screen;
4. Call this device to light up the screen and ring

In this case, the device will stop ringing for 1 minute after receiving an incoming call, and then a prompt will pop up on the screen.
**After the prompt pops up, the screen still cannot be turned off. Unless the battery is exhausted or manually terminated after being discovered by the user. For details, see #7107**

## Occasion 2:
1. Make the Element appear on the screen;
2. Press the Recent Task key to remove Element;
3. Press the power button to turn off and lock the screen;
4. Call the device, causing it to light up the screen and ring.

In this case, the ringing will continue for 1 minute after the device receives an incoming call, and the screen will stay on until the battery is exhausted or manually terminated after being discovered by the user.

----
I wanted to take some time to make a video recreating the above occasion.
**Battery optimization has been lifted for Element in the above two situations, and Element is allowed to run persistently when the screen is locked. Element version is 1.6.5, and the push method is self-hosted ntfy.**
